tutorials/2d/2d_movement.rst

```diff
@@ -0,0 +1,207 @@
+cancel each other out, but will also result in diagonal movement being faster
+due to the two directions being added together.
+
+We can prevent that if we *normalize* the velocity, which means we set
+its *length* to ``1``, and multiply by the desired speed.
+
+.. tip:: If you've never used vector math before, or just need a refresher,
+         you can see an explanation of vector usage in Godot at :ref:`doc_vector_math`.
+
+
+
+
+
+Rotation + Movement
+-------------------
+
+This type of movement is sometimes called "Asteroids-style" because it resembles
+how that classic arcade game worked. Pressing left/right rotates the character,
+while up/down moves it forward or backward in whatever direction it's facing.
+
+.. image:: img/movement_rotate1.gif
+
+:: 
+    
+    extends KinematicBody2D
+
+    export (int) var speed = 200
+    export (float) var rot_speed = 1.5
+
+    var velocity = Vector2()
+    var rot_dir = 0
+
+    func get_input():
+        rot_dir = 0
+        velocity = Vector2()
+        if Input.is_action_pressed('right'):
+            rot_dir += 1
+        if Input.is_action_pressed('left'):
+            rot_dir -= 1
+        if Input.is_action_pressed('down'):
+            velocity = Vector2(-speed, 0).rotated(rotation)
+        if Input.is_action_pressed('up'):
+            velocity = Vector2(speed, 0).rotated(rotation)
+
+    func _physics_process(delta):
+        get_input()
+        rotation += rot_dir * rot_speed * delta
+        move_and_slide(velocity)
+        
+Here we've added two new variables to track our rotation direction and speed.
+Again, pressing both keys at once will cancel out and result in no rotation.
+The rotation is applied directly to the body's ``rotation`` property.
+
+To set the velocity, we use the ``Vector2.rotated()`` method so that it points
+in the same direction as the body. ``rotated()`` is a very useful vector function
+that you can use in many circumstances where you would otherwise need to apply
+trigonometric functions.
+
+Rotation + Movement (mouse)
+---------------------------
+
+This style of movement is a variation of the previous one. This time, the direction
+is set by the mouse position instead of the keyboard. The character will always
+"look at" the mouse pointer. The forward/back inputs remain the same, however.
+
+.. image:: img/movement_rotate2.gif
+
+::
+    
+    extends KinematicBody2D
+
+    export (int) var speed = 200
+
+    var velocity = Vector2()
+
+    func get_input():
+        look_at(get_global_mouse_position())
+        velocity = Vector2()
+        if Input.is_action_pressed('down'):
+            velocity = Vector2(-speed, 0).rotated(rotation)
+        if Input.is_action_pressed('up'):
+            velocity = Vector2(speed, 0).rotated(rotation)
+
+    func _physics_process(delta):
+        get_input()
+        move_and_slide(velocity)
+
+Note that the :ref:`Node2D <class_Node2D>` ``look_at()`` method exists just
```

Comment:
Sounds a little convoluted. We can use look_at to make the character/ship rotate towards the mouse

Reply:
Reworded